File: Back/src/main/java/com/ssafy/myalbba/common/entity/Store.java
package com.ssafy.myalbba.common.entity;

import com.ssafy.myalbba.Member.entity.Owner;
import com.ssafy.myalbba.common.dto.StoreDTO;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Store {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "store_id")
    private Long id;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "owner_id", referencedColumnName = "id", nullable = false)
    private Owner owner;

    private String name;
    private String address;
    private String phoneNumber;
    private int payday;

    private Boolean reviewWidget;

    @PrePersist
    private void prePersist(){
        if(this.reviewWidget == null){ this.reviewWidget = false; }
    }

    public void setOwner(Owner owner) { this.owner = owner; }

    private Store (String name, String address, String phoneNumber, int payday) {
            this.name = name;
            this.address = address;
            this.phoneNumber = phoneNumber;
            this.payday = payday;
    }

    public static Store createStore(String name, String address, String phoneNumber, int payday) {
        return new Store(name, address, phoneNumber, payday);
    }

    public Store(long id, String name, String address, String phoneNumber, int payday) {
        this.id = id;
        this.name = name;
        this.address = address;
        this.phoneNumber = phoneNumber;
        this.payday = payday;
    }

    public void updateStore(StoreDTO storeDTO) {
        this.name = storeDTO.getName();
        this.address = storeDTO.getAddress();
        this.phoneNumber = storeDTO.getPhoneNumber();
        this.payday = storeDTO.getPayday();
    }

    public void setReviewWidget(boolean reviewWidget) { this.reviewWidget = reviewWidget; }
}

File: Front/src/common/apis/api/store.js
import axiosInstance from '@/common/apis/utils/axiosInstance';
import { API_URLS } from '@/common/apis/utils/apiConfig';

// {
// 	"name":"멀티캠퍼스 역삼",
// 	"address":"서울 강남구 테헤란로 212",
// 	"phoneNumber":"02-3429-5114",
// 	"payday":"10",
// }
//가게 정보 등록
export const registerStore = async (data) => {
  const response = await axiosInstance.post(API_URLS.STORE, data);
  return response.data;
};

// 가게 정보 조회
export const getStore = async () => {
  const response = await axiosInstance.get(API_URLS.STORE);
  return response.data;
};

//가게 정보 수정
// {
// 	"name":"멀티캠퍼스 역삼",
// 	"address":"서울 강남구 테헤란로 212",
// 	"phoneNumber":"02-3429-5114",
// 	"payday":"10",
// }
export const updateStore = async (data) => {
  const response = await axiosInstance.put(API_URLS.STORE, data);
  return response.data;
};
